Upgraded built-in libraries to support sdk34 and material3

waelchateur opened the pull request:
** Updated ** 

activity 1.6.1
material 1.12
appcompat 1.6.1
appcompat-resources 1.6.1
browser 1.7.0
constraintlayout 2.1.4
constraintlayout-core 1.0.4
androidX core 1.9.0
core-ktx 1.8.0
dynamic-animation-1.1.0
emoji2-1.2.0
emoji2-views-helper-1.2.0
lifecycle-common-2.5.1
lifecycle-runtime-2.5.1
lifecycle-process-2.4.1
lifecycle-viewmodel-2.6.1
lifecycle-viewmodel-savedstate-2.6.1
media-1.7.0
recyclerview-1.3.2
resourceinspection-annotation-1.0.1
savedstate-1.2.1
startup-runtime-1.1.1
transition-1.5.0
kotlin-stdlib-1.6.21
kotlin-stdlib-jdk7-1.6.21

                       ** Added **

constraintlayout-solver 2.0.4
recyclerview-selection-1.1.0
customview-poolingcontainer:1.0.0

framework has been updated to sdk34

ilyassesalama commented:
Can you guys please look into this. Especially @hasrat-ali since you've done this kind of change before.

hasrat-ali reviewed (commented):
First of all nice contribution.

hasrat-ali reviewed (changes requested):
Why there is different libs.zip and dexs.zip files for both varients (`minApi21`, `minApi26`)?

hasrat-ali commented on app/src/main/java/a/a/a/Ix.java:
```diff
@@ -239,7 +239,6 @@ private void writeAndroidxRoomService(XmlBuilder application) {
 
     private void writeAndroidxStartupInitializationProvider(XmlBuilder application) {
         var initializers = Set.of(
-                new Pair<>(builtInLibraryManager.containsLibrary(BuiltInLibraries.ANDROIDX_EMOJI2), "androidx.emoji2.text.EmojiCompatInitializer"),
```
Why you removed this library?

waelchateur replied:
its the first thing i did cause this didnt allow me to investigate more so i remove it from being initialised, more i digged the net about it and i found out that this will make the created apps more smoother for example scroling contents in recycler and etc more i didnt see any impact to projects rather than that user still can use the lib as he want

waelchateur replied:
i u revert this, app will crash without any log just leave it as is for now we can investigate it later

waelchateur replied:
@hasrat-ali still want this to be reverted ?

hasrat-ali commented on app/src/minApi26/java/mod/jbk/build/BuiltInLibraries.java:
```diff
@@ -0,0 +1,614 @@
+    public static String JETBRAINS_ANNOTATIONS = "annotations-13.0";
+    public static String KOTLIN_STDLIB = "kotlin-stdlib-1.6.21";
+    public static String KOTLIN_STDLIB_JDK7 = "kotlin-stdlib-jdk7-1.6.21";
+    public static String KOTLIN_STDLIB_JDK8 = "kotlin-stdlib-jdk8-1.6.21";
```
and this is just unnecessary because it's don't have any class in

waelchateur replied:
i see many internal libs have the jdk8 as a dependency so i add it do u recommend removing it ??

waelchateur replied:
> and this is just unnecessary because it's don't have any class in

as i can see it does have u can check dex, and since its an addon so the main kotlin core use it and as i told u before many libs have that as a dependencie.

waelchateur replied:
reverted ✅️

hasrat-ali commented:
> minsdk21 is not impacted with this update

why???????

hasrat-ali commented on app/src/minApi26/java/mod/jbk/build/BuiltInLibraries.java:
```diff
@@ -0,0 +1,614 @@
+            "androidx.customview.poolingcontainer"),
+            new BuiltInLibrary(ANDROIDX_CUSTOMVIEW, List.of(ANDROIDX_ANNOTATION, ANDROIDX_CORE)),
+            new BuiltInLibrary(ANDROIDX_RECYCLERVIEW_SELECTION),
+            new BuiltInLibrary(ANDROIDX_RECYCLERVIEW, List.of(ANDROIDX_ANNOTATION, ANDROIDX_COLLECTION, ANDROIDX_CORE, ANDROIDX_LEGACY_SUPPORT_CORE_UI, ANDROIDX_CUSTOMVIEW, ANDROIDX_RECYCLERVIEW_SELECTION, ANDROIDX_CUSTOMVIEW_POOLING_CONTAINER),
```
RecyclerView Selection library isn't dependency of RecyclerView. why you add this as recyclerview dependency?

waelchateur replied:
its a recyclerview addon and i think its usefull for enhancing the multi item selection even make it simpler so why not to concider it as a recycler dependencie

waelchateur commented:
> Why there is different libs.zip and dexs.zip files for both varients (`minApi21`, `minApi26`)?

here i can blame google for that, cause the upgrading to newer libs break the minsdk21, users cant compile even simple projects so i did that to keep the minsdk21 impacted .

waelchateur commented:
> > minsdk21 is not impacted with this update
> 
> why???????

users complain a lot about the update and they got plenty of errors even they cant compile a simple project because of the material lib update ,more i dont have any device to debug those errors . i have concider the minsdk21 as a future project for now.

waelchateur commented:
the add material3 switch and the dynamic theme by aquiles encorage me to do this Pr and since he reverted his features this Pr is useless for now  since every material 3 component or widget or style will make the app crash if project is not using m3 theme 🫤

RofrancoD commented:
they could put 2 switches

ilyassesalama commented:
@trindadedev13 Can you open your pull request again so we can do some proper testing? Also why did you close your PR?

elfilibusterismo commented:
> > @trindadedev13 Can you open your pull request again so we can do some proper testing? Also why did you close your PR?
> 
> what PR ?

The material3 thing

trindadedev13 commented:
> > > @trindadedev13 Can you open your pull request again so we can do some proper testing? Also why did you close your PR?
> > 
> > 
> > what PR ?
> 
> The material3 thing

ah,  I didn't make a PR for this.

trindadedev13 commented:
Why closed?

subhash16 commented:
Why closed?

waelchateur commented:
> Why closed?

because no boddy have some interrest in it and now it has conflicts, i can see everyone pushin ui stuff but this naaa we dont care , iliyasse said "What’s blocking us from merging it?" no one answer and obviously it was nothing the Pr is very clear just one class to be reviewed and actualy was reviewed by hasrat ali and i explained all , then iliyasse said "He made critical changes that must be tested well before merged" wtf i passed many nights testing and fixing and didnt make a Pr until make sure that everything 100% working, more i did some examples so the community can test the apk with me , i was very happy with the result but no one care its seems not importing so i closed the Pr

ilyassesalama commented:
I hate when people expect their PR to be merged the same day they opened their PR. Even if you spend years fixing bugs and making sure everyone works well, we need to review them. Not everyone’s perfect, everyone can make mistakes in their code. My point is clear, no need for further explanation.

waelchateur commented:
> I hate when people expect their PR to be merged the same day they opened their PR. Even if you spend years fixing bugs and making sure everyone works well, we need to review them. Not everyone’s perfect, everyone can make mistakes in their code. My point is clear, no need for further explanation.

i didnt object, but this pr already reviewed its very simple just one class and it has 4 days now waiting & every new pr got accepted but not this more if u can see the new changes will make this pr usless ,m3 theme switch got removed, dynamic theme too & sadly now it has conflicts, this pr should be the first one accepted so new contributions could be based on these changes

waelchateur commented:
Ready to be merged :

fixed conflicts , reverted adding kotlin-stdlib-jdk8-1.6.21, reverted the built-in libs separation can be added later since its the only fix for minsdk21 since no more updates needed for old phones upgrades will make sketchware a bugware for them so u can deside, the emoji2 init thing cant be debuged from my side & i did the best option by disable it. any suggestion would be appreciated 🤝

hasrat-ali commented on app/src/main/java/mod/jbk/build/BuiltInLibraries.java:
```diff
@@ -114,7 +114,6 @@ public class BuiltInLibraries {
     public static String JETBRAINS_ANNOTATIONS = "annotations-13.0";
     public static String KOTLIN_STDLIB = "kotlin-stdlib-1.6.21";
     public static String KOTLIN_STDLIB_JDK7 = "kotlin-stdlib-jdk7-1.6.21";
```
You should remove too

waelchateur replied:
its already exist in sketchware but ok why not i will remove it

waelchateur replied:
done ✅️

PranavPurwar commented on app/src/main/java/mod/jbk/build/BuiltInLibraries.java:
```diff
@@ -290,19 +300,17 @@ public class BuiltInLibraries {
-            new BuiltInLibrary(KOTLIN_STDLIB_JDK7, List.of(KOTLIN_STDLIB)),
             new BuiltInLibrary(LOTTIE, List.of(ANDROIDX_APPCOMPAT, OKIO), "com.airbnb.lottie"),
-            new BuiltInLibrary(MATERIAL, List.of(ANDROIDX_ANNOTATION, ANDROIDX_ANNOTATION_EXPERIMENTAL, ANDROIDX_APPCOMPAT, ANDROIDX_CARDVIEW,
+            new BuiltInLibrary(MATERIAL, List.of(ANDROIDX_ANNOTATION, ANDROIDX_ANNOTATION_EXPERIMENTAL, ANDROIDX_ACTIVITY, ANDROIDX_RESOURCEINSPECTION_ANNOTATION, ANDROIDX_APPCOMPAT, ANDROIDX_CARDVIEW,
```
That resourceinspection dependency is not useful in sketchware. It's only used by android studio. You can remove it.

waelchateur commented:
> Why there is different libs.zip and dexs.zip files for both varients (`minApi21`, `minApi26`)?

already reverted this

hasrat-ali reviewed (changes requested):
Bro, look, I don't want to keep you in the shadow.
 But I don't think that this pull request will be merged.
Because, we want to make it in such a way that we do not have to do too much hard work to update the libraries (in future)

waelchateur commented:
> Bro, look, I don't want to keep you in the shadow. But I don't think that this pull request will be merged. Because, we want to make it in such a way that we do not have to do too much hard work to update the libraries (in future)

thank you for your honesty

subhash16 commented:
Big update closed

waelchateur commented:
> Bro, look, I don't want to keep you in the shadow. But I don't think that this pull request will be merged. Because, we want to make it in such a way that we do not have to do too much hard work to update the libraries (in future)

thank you for your honesty 

> Big update closed

sorry bro ❤️
